tests/test_scan_local_repo.py

```diff
@@ -28,6 +28,27 @@ def test_scan_exits_gracefully_when_target_is_not_git_repo(self):
                 f"{Path(dirname).resolve()} is not a valid git repository.\n",
             )
 
+    def test_scan(self):
+        runner = CliRunner()
+        with runner.isolated_filesystem() as dirname:
+            repo_path = util.clone_git_repo(
+                "https://github.com/godaddy/tartufo.git", target_dir=dirname
+            )
+            result = runner.invoke(cli.main, ["scan-local-repo", repo_path.as_posix()])
+        self.assertLess(result.exit_code, 2)
+
+    def test_scan_branch(self):
+        runner = CliRunner()
+        with runner.isolated_filesystem() as dirname:
+            repo_path = util.clone_git_repo(
+                "https://github.com/godaddy/tartufo.git", target_dir=dirname
+            )
```

Comment:
I'm very much not a fan of this, for a couple of reasons. This changes our test suite from being pretty strict unit tests, to relying on external services. It's also going to greatly increase the run time for the test suite, since we're now running two full scans of this project on every test run. This was something that was causing pretty big problems before and actually just recently got fixed (#55).

I think that we should instead use some well thought-out mocks and assertions on those mocks. I would be happy to work with you on setting those up if you would like. 😄